Plot kwargs

dpanici opened the pull request:
Documents kwargs for plotting functions and adds sensible kwargs to those that were lacking them (such as figsize)
 - Document kwargs, according to [matplotlib-style documentation](https://stackoverflow.com/questions/62511086/how-to-document-kwargs-according-to-numpy-style-docstring)
 - add kwargs to plotting functions missing sensible/useful kwargs
 - add check for unused kwargs to most plotting functions
 - add plot comparisons to artifact uploaded to github 
 - bump `pytest-mpl` version to fix [bug](https://github.com/matplotlib/pytest-mpl/pull/171) where if `unittest.TestCase` is used, it did not actually run those tests correctly 

resolves #178 , #244

dpanici commented:
failing only code coverage tests, but adding tests to cover just the lines that check if a kwarg has been passed adds unnecessary time to the tests for something that is not needed to test multiple times imo

f0uriest commented on desc/plotting.py:
```diff
@@ -492,21 +531,24 @@ def plot_2d(eq, name, grid=None, log=False, norm_F=False, ax=None, **kwargs):
     contourf_kwargs["extend"] = "both"
 
+    if len(kwargs):
+        raise ValueError(f"plot_2d got unexpected keyword argument: {kwargs.keys()}")
```
a better way to handle this might be ``assert len(kwargs)==0, msg``, that way it will always be checked by the testing

f0uriest commented:
Some other things that would be nice to have as settable parameters via kwargs:
- font size for axis labels, titles etc
- line widths and line styles and colors for plot_1d, plot_fsa (similar to plot_surfaces)
- possibly orientation of 3d plots?

dpanici commented:
- [x] title font size
 - [ ] axes label font sizes
 - [ ] line widths, styles and colors for plot_1d and plot_fsa
 - [x] 3-D plot orientation

f0uriest commented on desc/plotting.py:
```diff
@@ -1,3 +1,4 @@
+from turtle import color
```
is this used?

dpanici replied:
nope I'll remove, not sure why it's there

f0uriest commented on desc/plotting.py:
```diff
@@ -298,6 +299,16 @@ def plot_coefficients(eq, L=True, M=True, N=True, ax=None):
+
+        Valid keyword arguments are:
+
+        figsize: tuple of length 2, the size of the figure (to be passed to matplotlib)
```
should add other kwargs to these

f0uriest commented on desc/plotting.py:
```diff
@@ -398,20 +429,33 @@ def plot_1d(eq, name, grid=None, log=False, ax=None, **kwargs):
+        )
     else:
-        ax.plot(grid.nodes[:, plot_axes[0]], data, label=kwargs.get("label", None))
+        ax.plot(grid.nodes[:, plot_axes[0]], data, label=kwargs.pop("label", None))
```
looks like some of the kwargs aren't used in this case?

f0uriest commented:
Do we want to also add an upload of the plot comparisons?

f0uriest reviewed (dismissed):
Looks fine. The image diffs aren't in the artifact but I'm pretty sure that's because the action always uses the yml from master, so we might need to debug the uploading a bit but fine to merge for now.

ddudt commented on desc/plotting.py:
```diff
@@ -2153,11 +2464,22 @@ def plot_field_lines_sfl(eq, rho, seed_thetas=0, phi_end=2 * np.pi, ax=None, **k
+        Toroidal angle to integrate field line from, in radians. Default is 0.
     phi_end: float
         Toroidal angle to integrate field line until, in radians. Default is 2*pi.
+    phi_end: float
```
Should this be `phi_spacing` or something similar? And this input should be used to set the `dphi` variable, which is currently undefined.

dpanici replied:
thanks, this should be `dphi` and `dphi=1e-2` is the default value (from the function signature)

dpanici commented:
> Looks fine. The image diffs aren't in the artifact but I'm pretty sure that's because the action always uses the yml from master, so we might need to debug the uploading a bit but fine to merge for now.

yea I think I fixed this, I had set the output path but by default things only get saved if tests fail. Changed so that it outputs a summary even if tests all pass

dpanici commented:
have written approval thru message to merge
